[REL-1004] Release notes for v23.2.7 (#18703)
cockroachdb/docs: src/current/_includes/releases/v23.2/v23.2.7.md

## v23.2.7

Release Date: July 2, 2024

{% include releases/new-release-downloads-docker-image.md release=include.release %}

<h3 id="v23-2-7-{{-site.data.products.enterprise-}}-edition-changes">{{ site.data.products.enterprise }} edition changes</h3>

> - [Changefeeds]({% link v23.2/change-data-capture-overview.md %}) now defaults to using the bulk oracle for planning, which distributes work evenly across all replica in the locality filter, including followers if enabled. To disable this behavior and use the previous bin-packing oracle, set the [cluster setting]({% link v23.2/cluster-settings.md %}) `changefeed.balanced_distribution.enabled` to `false`. [#124925][#124925]

Reviewer: ```suggestion
- [Changefeeds]({% link v23.2/change-data-capture-overview.md %}) now default to using the bulk oracle for planning, which distributes work evenly across all [replicas]({% link v23.2/architecture/reads-and-writes-overview.md %}#replica) in the locality filter, including followers if enabled. To disable this behavior and use the previous bin-packing oracle, set the [cluster setting]({% link v23.2/cluster-settings.md %}) `changefeed.balanced_distribution.enabled` to `false`. [#124925][#124925]
```

> - [`ALTER CHANGEFEED`]({% link v23.2/alter-changefeed.md %}) no longer removes the CDC query when [modifying changefeed properties]({% link v23.2/create-and-configure-changefeeds.md %}#modify-a-changefeed). [#125437][#125437]

Reviewer: ```suggestion
- [`ALTER CHANGEFEED`]({% link v23.2/alter-changefeed.md %}) no longer removes the [CDC query]({% link v23.2/cdc-queries.md %}) when modifying changefeed properties. [#125437][#125437]
```

> - The new metric `changefeed.usage.table_byte` allows you to observe billing for changefeed usage in CockroachDB {{ site.data.products.cloud }}. [#125525][#125525]

Reviewer: ```suggestion
```

I need to let the team know that we should remove any release notes relating to billing for now.

<h3 id="v23-2-7-sql-language-changes">SQL language changes</h3>

- Precision is no longer limited when encoding `geo` data types to JSON. [#124535][#124535]

> - When the new `optimizer_push_offset_into_index_join` [session setting]({% link v23.2/set-vars.md %}) is enabled, the optimizer attempts to produce more efficient query plans by attempting to push offset expressions into index join expressions to produce more efficient query plans. [#124492][#124492]

Reviewer: ```suggestion
- When the new `optimizer_push_offset_into_index_join` [session setting]({% link v23.2/set-vars.md %}) is enabled, the [optimizer]({% link v23.2/cost-based-optimizer.md %}) attempts to produce more efficient query plans by attempting to push offset expressions into index join expressions to produce more efficient query plans. [#124492][#124492]
```

<h3 id="v23-2-7-general-changes">General changes</h3>

- CockroachDB v23.2.7 and subsequent v23.2 releases are eligible for [long term support (LTS)]({% link releases/release-support-policy.md %}#support-types).

<h3 id="v23-2-7-operational-changes">Operational changes</h3>

> - Improved metrics related to disk usage reporting for volumes that dynamically change their size over time. [#125107][#125107]

Reviewer: ```suggestion
- Improved metrics related to [disk usage]({% link v23.2/ui-storage-dashboard.md %}#capacity-metrics) reporting for volumes that dynamically change their size over time. [#125107][#125107]
```

<h3 id="v23-2-7-security-changes">Security changes</h3>
- Improved the automated cleanup when dropping roles inside descriptors. [#124665][#124665]

<h3 id="v23-2-7-bug-fixes">Bug fixes</h3>

> - Fixed a bug where a range with a replication factor of `1` to be scaled up to a replication factor of `2`.

Author: ```suggestion
- Fixed a bug where a range with a replication factor of `1` to be scaled up to a replication factor of `2`. [#124487][#124487]
```

- Fixed a [changefeed]({% link v23.2/create-and-configure-changefeeds.md %}) panic in v24.1, v23.2, and v23.1 when the [cluster setting]({% link v23.2/cluster-settings.md %}) `changefeed.aggregator.flush_jitter` [cluster setting]({% link v23.2/cluster-settings.md %}) is configured and a changefeed's `min_checkpoint_frequency` option is set to zero. [#125469][#125469]
- Fixed a bug where the public schema was erroneously created with its owner set to the `admin` role instead of the database owner.Ownership  of the public schema can be altered after the schema is created. [#125535][#125535]
- Fixed a bug introduced in v23.2.0 where inserting rows into a [`REGIONAL BY ROW` table]({% link v23.2/table-localities.md %}#regional-by-row-tables) could cause an internal error if the source was a `VALUES` clause with a single row and at least one boolean expression. [#126208][#126208]

<h3 id="v23-2-7-performance-improvements">Performance improvements</h3>

- The optimizer now generates more efficient query plans for some queries with [`OFFSET`]({% link v23.2/limit-offset.md %}#offset) clauses. [#124492][#124492]

> 

Author: I did, because I thought we decided we are removing them now. But maybe it was just the contributors, so I'll put it back.

> 

Author: ```suggestion

<div class="release-note-contributors" markdown="1">

<h3 id="v23-2-7-contributors">Contributors</h3>

This release includes 77 merged PRs by 30 authors.
</div>
```
